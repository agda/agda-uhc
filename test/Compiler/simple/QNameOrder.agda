
module _ where

open import Agda.Builtin.Reflection
open import Agda.Builtin.Bool
open import Agda.Builtin.Unit
open import Agda.Builtin.String
open import Common.Prelude

_<_ = primQNameLess

True : Bool → Set
True true  = ⊤
True false = ⊥

zzz aaa : ⊤
zzz = _
aaa = _

⊥-elim : {A : Set} → ⊥ → A
⊥-elim ()

check : (x y : Name) → True (x < y) → String
check x y prf with x < y
check x y prf | true = "A-ok"
check x y prf | false = ⊥-elim prf

main : IO Unit
main = putStrLn (check (quote zzz) (quote aaa) _)
