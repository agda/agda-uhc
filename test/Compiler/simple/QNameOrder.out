EXECUTED_PROGRAM

ret > ExitSuccess
out > A-ok
out >
